src/hooks/useCheckEthBalance.js

```diff
@@ -0,0 +1,23 @@
+import { useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import { hasEthBalance } from '../handlers/web3';
+import { setIsWalletEthZero } from '../redux/isWalletEthZero';
+import { logger } from '../utils';
+import useAccountSettings from './useAccountSettings';
+
+export default function useCheckEthBalance() {
+  const dispatch = useDispatch();
+  const { accountAddress } = useAccountSettings();
+
+  const check = useCallback(async () => {
+    try {
+      const ethBalance = await hasEthBalance(accountAddress);
+      const isZero = !ethBalance;
+      dispatch(setIsWalletEthZero(isZero));
+    } catch (error) {
+      logger.log('Error: Checking eth balance', error);
+    }
+  }, [dispatch, accountAddress]);
+
+  return check;
```

Comment:
you dont need to change this, but you could just directly return the `useCallback` instead of assigning it to `const check`